docs/visual_parity.md

```diff
@@ -0,0 +1,56 @@
+# Visual Parity
+
+Quilla includes a special `XPath` validation state called `VisualParity`. This page describes what `VisualParity` is in detail, how it can be used for UI regression tests, and how it integrates storage plugins to manage the images it uses and creates. Quilla strives to make the handling of auxiliary data (baseline images, etc) needed by VisualParity as simple as possible.
+
+## What is VisualParity?
+
+VisualParity is a validation state that uses previously accepted screenshots generated by Quilla ("baseline images") to compare against screenshots generated when Quilla is run during testing ("treatment images") to determine if the new screenshots match the old ones. By using VisualParity with small sections of a web page, Quilla can give assurances that each section remains the same as it has been from previous runs, allowing users to update only the baseline images related to the work they are doing and protecting the rest of the page from having unexpected visual changes introduced.
+
+## Configuring a Storage Plugin
+
+VisualParity requires the ability to store and use images, many of which are required to persist between runs. To do so, it uses one of a few possible "storage plugins"- Quilla Plugins that handle storing, organizing, retrieving, and potentially deleting images. This allows Quilla to simplify the validation logic by outsourcing the image handling at key points to an outside mechanism.
+
+Quilla does not enforce any specific method of configuration for the storage plugins, opting to leave this to plugin authors. To configure an external Quilla storage plugin, please refer to the documentation of that plugin.
+
+Quilla has bundled two storage plugins- a LocalStorage and a BlobStorage plugin. Both of these plugins are configured through CLI that can be reviewed by running `quilla --help`. The LocalStorage plugin requires the specification of a directory in which to store images, and the BlobStorage requires an Azure Blob Storage connection string to connect to a Cloud storage container.
+
+## Creating/Updating Baseline Images
+
+Quilla includes options to create/update baseline images on a per-baseline and a per-file basis. This is done to allow tests to be written that contain multiple `VisualParity` validations while maintaining confidence that updating the baseline image for a specific validation will not create a false positive in any other validation.
```

Comment:
do you think "per-test-suite" is a better term to replace "per-file"?

Reply:
I don't think so because "per-test-suite" implies that it will generate baseline images for multiple Quilla files in one go, which Quilla doesn't. Technically, one *can* do it at the test-suite level but if and only if they are using it with Pytest since Quilla does not allow one to run a full suite with its CLI (you can do that using either pytest or by making a script such as `find tests -type f -name "test_*.json" -exec quilla -f {} \;`)